app/Http/Controllers/Admin/ModulesController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Contracts\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Artisan;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Str;
+use Laracasts\Flash\Flash;
+use Madnest\Madzipper\Madzipper;
+use App\Models\Module;
+
+class ModulesController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @param Request $request
+     *
+     * @return mixed
+     */
+    public function index()
+    {
+        $modules = Module::all();
+        return view('admin.modulesmanager.index', [
+            'modules' => $modules
+        ]);
+    }
+
+    public function addModule(Request $request)
+    {
+        if ($request->isMethod('get')) {
+            return view('admin.modulesmanager.add');
+        }
+        elseif ($request->isMethod('post')) {
+            $this->addModulePost($request);
+        }
+        Flash::success('Module Installed Successfully!');
+        return redirect(route('admin.modulesmanager.index'));
+    }
+
+    public function addModulePost($request)
+    {
+        $zipper = new Madzipper;
```

Comment:
This code should be refactored and moved to a service in `app/Services/ModuleService.php`. This makes it easier to add unit tests in